botocore/utils.py

```diff
@@ -554,6 +558,12 @@ def _log_imds_response(self, response, reason_to_log, log_body=False):
             logger_args.append(response.content)
         logger.debug(statement, *logger_args)
 
+    def _check_imdsv1_disabled(self):
+        if self._imds_v1_disabled:
```

Comment:
I think we should be a little more explicit here. What if someone called this class directly and set  `ec2_metadata_v1_disabled` to `foo`?

Reply:
Both are private variables, so I think we can say that's outside of the scope of expected behavior. We're already ensuring the variable we assign is a boolean, if they decide to mutate it, then here be dragons.

Reply:
It looks like we only coerce to a boolean for `self._disabled` not `self._imds_v1_disabled`

Reply:
We should be using the same logic for both. Deviating where/how we check is going to create more pain in the future.